modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/exp/ConverterUtils.java

```diff
@@ -277,6 +277,17 @@ public static Expression convert(Expression operand, Type fromType, Type toType)
 
             if (fromPrimitive != null) {
                 // E.g. from "float" to "double"
+                // check overflow for 'integer' subtypes
```

Comment:
I'm afraid this cover only non boxed privitives. In case of boxed primitives we will get something like  `longVal.intValue()` without any checks.